routes/helperMethods.js

```diff
@@ -21,6 +22,7 @@ function errorHandler(res, status, err) {
 }
 
 async function distributionCreditUpdate(distribution, course, add) {
+  if (!distribution) return; 
```

Comment:
this function was triggering `cannot read field planned from null`, hence the return if distribution is null. As to why distribution might be null? We're not entirely sure. My guess is that we're setting course.distribution_ids wrong.

Reply:
I'm thinking this could be a legacy issue where in past operations, distribution ids were not deleted properly from the ids field of the course but the actual distribution object was deleted, therefore it cannot find the object with the id. This issue should not happen in the future once we've cleared all deleted distributions ids and have the correct operations in place.

Reply:
No, this is a new issue with new code added by Minseo and Akhil. The error occurred in a helper method they wrote I think, used within the for loop.

Reply:
I think? or maybe not, you're the authority on this I guess @QinyuanWu since it may have been old code or new code.

Reply:
I'm not sure about the new helper method but current courses definitely contains removed distribution ids.